src/api/ledger/parse/orderbook-order.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+const _ = require('lodash');
+const utils = require('./utils');
+const flags = utils.core.Remote.flags.offer;
+const parseAmount = require('./amount');
+
+function parseOrderbookOrder(order: Object): Object {
+  const direction = (order.Flags & flags.Sell) === 0 ? 'buy' : 'sell';
+  const takerGetsAmount = parseAmount(order.TakerGets);
+  const takerPaysAmount = parseAmount(order.TakerPays);
+  const quantity = (direction === 'buy') ? takerPaysAmount : takerGetsAmount;
+  const totalPrice = (direction === 'buy') ? takerGetsAmount : takerPaysAmount;
+
+  const specification = utils.removeUndefined({
+    direction: direction,
+    quantity: quantity,
+    totalPrice: totalPrice,
+    passive: ((order.Flags & flags.Passive) !== 0) || undefined
+  });
+  // "quality" is omitted intentionally as it corresponds to
```

Comment:
> Incorrect. 

I meant that "totalPrice" is the only one that is correct out of "price" and "totalPrice". "quality" is correct, but it is specific to rippled, unlike "price", which everyone understands. And I'm not sure if it's any more correct to say that orders get executed at the rate of the "quality" than it is to say they get executed at the ratio specified by "TakerGets/TakerPays". "quality" is basically an intermediate result of the calculation for partial executions and you still have to multiply by the quantity traded in order to get the final result, unless the quantity is exactly 1.0. You could argue that including the intermediate result might reduce the total accumulated error, but it won't (I think) if the client code uses the `IOUValue` class to calculate the partial execution price.

> I didn't really understand that.

If a user sees a result that contains "price" and "totalPrice" they are liable to assume that they can use either one safely, whereas this would not be true because "price" would be an approximation.

> What is this api going to be used for? 
> Who is going to use it? 
> What are they all going to need?

The design goal of this API is to be "the one Ripple API", meaning that ideally it should be able to replace all of rippled's APIs, ripple-lib, ripple-rest, and everything else. For REST servers, the API would have to be slightly different, but the schemas should be the same. Of course, for it to replace rippled's APIs it would have to be ported to a compilable language. But also we can't implement everything at once, so we're just trying to do the most important features first.

> Are all the orders sorted by price as they were before? 
> Is there expectations that they will be?

Yes, I forgot to sort the results, will fix.

> not all orders returned by rippled make it into the object returned by getOrderBook

That's correct. Semantically the request should only return "half" of the results. For convenience though, we may as well return the flipped order book since we already have all the results from rippled. I'll try to find a clean way to do that.